### PR TITLE
Added documentation to Detector library's public api

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: src/Detector/Contracts/IPlatformDetector.cs [hunk 1]
```diff
@@ -6,12 +6,12 @@
 namespace Microsoft.Oryx.Detector
 {
     /// <summary>
-    /// Detects language name and version of the application in source directory.
+    /// An abstraction to detect a platform and additional information of the application in the source directory.
```
Comment (gpcastro) [inline src/Detector/Contracts/IPlatformDetector.cs:9]: in _a_ source directory?

File: src/Detector/Contracts/IPlatformDetector.cs [hunk 2]
```diff
@@ -6,12 +6,12 @@
...
     /// </summary>
     public interface IPlatformDetector
     {
         /// <summary>
-        /// Detects language name and version of the application in source directory.
+        /// Detects platform and additional information of the application in the source directory.
```
Comment (gpcastro) [inline src/Detector/Contracts/IPlatformDetector.cs:14]: I'd remove "and additional information", as it is not clear what that means

File: src/Detector/DefaultPlatformDetector.cs [hunk 3]
```diff
@@ -8,11 +8,21 @@
 
 namespace Microsoft.Oryx.Detector
 {
+    /// <summary>
+    /// A default implementation of <see cref="IDetector"/> which invokes the
```
Comment (gpcastro) [inline src/Detector/DefaultPlatformDetector.cs:12]: _The_ default

File: src/Detector/DetectorContext.cs [hunk 4]
```diff
@@ -6,10 +6,13 @@
 namespace Microsoft.Oryx.Detector
 {
     /// <summary>
-    /// Context to create a detector.
+    /// The information which is used by the detectors to detect applications.
```
Comment (gpcastro) [inline src/Detector/DetectorContext.cs:9]: "Which is" is unnecessary

File: src/Detector/DotNetCore/DotNetCorePlatformDetectorResult.cs [hunk 5]
```diff
@@ -5,8 +5,18 @@
...
 namespace Microsoft.Oryx.Detector.DotNetCore
 {
+    /// <summary>
+    /// Represents a <see cref="PlatformDetectorResult"/> returned by the <see cref="DotNetCoreDetector"/> and
+    /// contains additional information related to the detected applications.
+    /// </summary>
     public class DotNetCorePlatformDetectorResult : PlatformDetectorResult
     {
+        /// <summary>
+        /// Relative path to the location of the project file.
```
Comment (gpcastro) [inline src/Detector/DotNetCore/DotNetCorePlatformDetectorResult.cs:15]: Gets or sets the...

File: src/Detector/PlatformDetectorResult.cs [hunk 6]
```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Oryx.Detector
...
     /// <summary>
     /// Represents the result of a <see cref="IPlatformDetector.Detect(DetectorContext)"/> operation.
+    /// This contains basic information of the detected application. However, each individual detector could include
+    /// additional information about the application. For example <see cref="DotNetCore.DotNetCoreDetector"/> returns
+    /// an instance of <see cref="DotNetCore.DotNetCorePlatformDetectorResult"/>.
     /// </summary>
     public class PlatformDetectorResult
     {
+        /// <summary>
+        /// The name the platform that was detected. For example: nodejs, dotnet, php and python.
```
Comment (gpcastro) [inline src/Detector/PlatformDetectorResult.cs:17]: Gets or sets the...

File: src/Detector/PlatformDetectorResult.cs [hunk 7]
```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Oryx.Detector
...
+        /// </summary>
         public string Platform { get; set; }
 
+        /// <summary>
+        /// The version of the platform that was detected.
```
Comment (gpcastro) [inline src/Detector/PlatformDetectorResult.cs:22]: Gets or sets the...

File: src/Detector/DotNetCore/DotNetCorePlatformDetectorResult.cs [hunk 8]
```diff
@@ -5,8 +5,18 @@
...
+        /// </summary>
+        /// <example>
+        /// src/ShoppingWeb/ShoppingWeb.csproj
```
Comment (qianz2) [inline src/Detector/DotNetCore/DotNetCorePlatformDetectorResult.cs:18]: nit: add quotes

File: src/Detector/Python/PythonDetector.cs [hunk 9]
```diff
@@ -11,15 +11,23 @@
...
 {
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects Python applications.
+    /// </summary>
     public class PythonDetector : IPythonPlatformDetector
     {
         private readonly ILogger<PythonDetector> _logger;
 
+        /// <summary>
+        /// Creates and instance of <see cref="PythonDetector"/>.
```
Comment (arroyc) [inline src/Detector/Python/PythonDetector.cs:22]: crates an instance **

File: src/Detector/Php/PhpDetector.cs [hunk 10]
```diff
@@ -11,15 +11,23 @@
...
 {
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects PHP applications.
+    /// </summary>
     public class PhpDetector : IPhpPlatformDetector
     {
         private readonly ILogger<PhpDetector> _logger;
 
+        /// <summary>
+        /// Creates and instance of <see cref="PhpDetector"/>.
```
Comment (arroyc) [inline src/Detector/Php/PhpDetector.cs:22]: creates an instance***

File: src/Detector/Hugo/HugoDetector.cs [hunk 11]
```diff
@@ -26,11 +29,16 @@ public class HugoDetector : IHugoPlatformDetector
             "theme",
         };
 
+        /// <summary>
+        /// Creates and instance of <see cref="HugoDetector"/>.
```
Comment (arroyc) [inline src/Detector/Hugo/HugoDetector.cs:33]: creates an instance ***

File: src/Detector/Python/PythonDetector.cs (hunk 9 above)
Comment (kichalla) [reply src/Detector/Python/PythonDetector.cs:22]: good catch!

File: src/Detector/LocalSourceRepo.cs [hunk 12]
```diff
@@ -12,36 +11,52 @@
 
 namespace Microsoft.Oryx.Detector
 {
+    /// <summary>
+    /// Default implementation of <see cref="ISourceRepo"/> which is backed by local file system.
```
Comment (gpcastro) [inline src/Detector/LocalSourceRepo.cs:15]: by _the_ local file system?

File: src/Detector/PlatformDetectorResult.cs [hunk 13]
```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Oryx.Detector
 {
     /// <summary>
     /// Represents the result of a <see cref="IPlatformDetector.Detect(DetectorContext)"/> operation.
+    /// This contains basic information of the detected application. However, each individual detector could include
```
Comment (gpcastro) [inline src/Detector/PlatformDetectorResult.cs:10]: basic information _about_ the detected...?

File: src/Detector/PlatformDetectorResult.cs [hunk 14]
```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Oryx.Detector
...
+    /// additional information about the application. For example <see cref="DotNetCore.DotNetCoreDetector"/> returns
+    /// an instance of <see cref="DotNetCore.DotNetCorePlatformDetectorResult"/>.
     /// </summary>
     public class PlatformDetectorResult
     {
+        /// <summary>
+        /// Gets or sets the name of the platform that was detected. For example: nodejs, dotnet, php and python.
```
Comment (gpcastro) [inline src/Detector/PlatformDetectorResult.cs:17]: Thinking about this from an API perspective, these properties should be immutable - we only set them in the constructor and whoever got the object can't change them

Comment (kichalla) [reply src/Detector/PlatformDetectorResult.cs:17]: Ok, let me file a work item to track this. I will handle it in a separate PR.

**Pull request merged**